File: benchmarks/matrix_multiply/matrix.py
N = 200

A = [i % 100 for i in range(N * N)]
B = [(i * 3 + 7) % 100 for i in range(N * N)]
C = [0] * (N * N)

for i in range(N):
    for j in range(N):
        s = 0
        for k in range(N):
            s += A[i * N + k] * B[k * N + j]
        C[i * N + j] = s

checksum = sum(C[i * N + i] for i in range(N))
print(checksum)

File: benchmarks/matrix_multiply/Matrix.java
public class Matrix {
    public static void main(String[] args) {
        int N = 200;

        int[] A = new int[N * N];
        int[] B = new int[N * N];
        int[] C = new int[N * N];

        for (int i = 0; i < N * N; i++) {
            A[i] = i % 100;
            B[i] = (i * 3 + 7) % 100;
        }

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                int sum = 0;
                for (int k = 0; k < N; k++) {
                    sum += A[i * N + k] * B[k * N + j];
                }
                C[i * N + j] = sum;
            }
        }

        int checksum = 0;
        for (int i = 0; i < N; i++) {
            checksum += C[i * N + i];
        }
        System.out.println(checksum);
    }
}
